app/models/purchase_phistory.rb

```diff
@@ -0,0 +1,31 @@
+# PurchaseモデルとPurchase-historyモデルのFormオブジェクト
+class PurchasePhistory
+  include ActiveModel::Model
+
+  attr_accessor :postal_code, :prefecture_id, :city, :house_number, :build_number, :phone_number,
+                # purchaseモデルの属性値
+                :item_id, :user_id,
+                # purchase_historyモデルの属性値
+                :token
+                # JSから送られてくるtoken情報
+
+  with_options presence: true do
+    validates :postal_code, format:{ with: /\A\d{3}[-]\d{4}\z/, message: "is invalid. Include hyphen(-) "}
+    # validates :postal_code, format:{ with: /\A\d{3}[-]\d{4}\z/ }
+    validates :city
+    validates :house_number
+    validates :phone_number, format:{ with: /[0-9]{10,11}/, message: "is invalid"}, length:{ in: 10..11 }
```

Comment:
```ruby
validates :phone_number, format:{ with: /[0-9]{10,11}/, message: "is invalid"}, length:{ in: 10..11 }
```

`format:{ with: /[0-9]{10,11}/, message: "is invalid"}`の記述ですでに桁数制限もできていますので、`, length:{ in: 10..11 }`の箇所は削除いただいて問題ないかと存じます。

削除後は念のため、ご自身で挙動の確認まで済ませていただきますようお願いいたします。

Reply:
ご指摘いただきありがとうございます。
当初、以下の正規表現でvalidationを設定していたのですが、phone_numberを11文字以上入力したとき、エラーコードが表示されなかったため、lengthを追記した経緯があります。
```suggestion
    validates :phone_number, format:{ with: /[0-9]{10,11}/ }
```

再度、正規表現の記述を見直し、以下のとおり記載したところ、11文字以上でもエラー表示することができました。
```suggestion
    validates :phone_number, format:{ with: /\A\d{10,11}\z/}
```

当初の表現でも、「数字が10~11桁繰り返し」となっていたと思いますが、Aとzを加えて、「初めから終わりまで、数字が10~11桁」としたことで、正しくvalidationされたのだと思います。